### PR TITLE
libmfu: add major, minor, patch version

### LINKED ISSUE #396: libmfu: make the shared library versioned
Looking at the files that are produced from the build I see:

```
/usr/lib64/libmfu.a
/usr/lib64/libmfu.so
```

The shared library looks unversioned.  There should be a libmfu.so._x_._y_ where the _x_ and _y_ are a version for the shared library, where the major version component (_x_) denotes ABI breakage that is non-backwards compatible.

Should this shared library be versioned as such?

### PR DESCRIPTION
**Pull request by daltonbohning**
Added the major, minor, and patch version
to the compiled libmfu.so

This is currently put into `src/common/CMakeLists.txt`, since that is where the `mfu` object is compiled.
If desired, we could put this in the root `CMakeLists.txt`

I don't have much experience with `cmake`, nor with the best practices here, so @adammoody I am okay with whichever you prefer.

Update: actually, I think it does make sense to keep this in `src/common/CMakeLists.txt`, since the _**package**_ and _**library**_ have separate versions.

closes #396 

Signed-off-by: Dalton Bohning <daltonx.bohning@intel.com>

### CODE REVIEW

File: src/common/CMakeLists.txt [hunk 1]
```diff
@@ -1,3 +1,9 @@
+# Version for the shared mfu library
+set(MFU_VERSION_MAJOR 0)  # Incompatible API changes
+set(MFU_VERSION_MINOR 11) # Backwards-compatible functionality
```
Comment (brianjmurrell) [inline src/common/CMakeLists.txt:3]: Starting the lib at 0.11.0?

**Review by brianjmurrell — COMMENTED** [review]
Generally speaking, the patch works and produces:

```
/usr/lib64/mpich/lib/libmfu.so.0.11.0
/usr/lib64/openmpi3/lib/libmfu.so.0.11.0
```

Just wonder if 0.11.0 is the correct starting place for versioning the library/API.

**Comment by adammoody** [comment]
Thanks for reminder about the library version number, @brianjmurrell .  I'm also good with ``0.0.0`` as a starting point.  Is that a valid number to use, or should we start with a 1.0.0?

**Review by brianjmurrell — APPROVED** [review]
0 is a typical starting point for shared libraries.

**Comment by daltonbohning** [comment]
@adammoody Does this look good to merge?

**Comment by adammoody** [comment]
Yes, go ahead and merge this in.  Thanks, @daltonbohning and @brianjmurrell .

**Pull request merged**